CommonTools/UtilAlgos/interface/ObjectSelectorProducer.h

```diff
@@ -0,0 +1,68 @@
+#ifndef CommonTools_UtilAlgos_ObjectSelectorProducer_h
+#define CommonTools_UtilAlgos_ObjectSelectorProducer_h
+
+#include "FWCore/Framework/interface/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/ConsumesCollector.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "CommonTools/UtilAlgos/interface/ParameterAdapter.h"
+#include "CommonTools/UtilAlgos/interface/StoreManagerTrait.h"
+#include "CommonTools/UtilAlgos/interface/SelectedOutputCollectionTrait.h"
+#include "CommonTools/UtilAlgos/interface/NullPostProcessor.h"
+#include "CommonTools/UtilAlgos/interface/EventSetupInitTrait.h"
+#include <utility>
+#include <vector>
+#include <memory>
+#include <algorithm>
+
+/**
+ * This class template is like ObjectSelector, but it is an EDProducer
+ * instead of EDFilter. Use case is that when the filter decisions are
+ * ignored (cms.ignore in configuration or EDFilter returns always
+ * true), EDProducers are better for the unscheduled mode than
+ * EDFilters.
+ */
+template<typename Selector,
+         typename OutputCollection = typename ::helper::SelectedOutputCollectionTrait<typename Selector::collection>::type,
+         typename PostProcessor = ::helper::NullPostProcessor<OutputCollection, edm::EDProducer>,
```

Comment:
Right, taking out the default values for the template parameters of ObjectSelectorProducer (as the stream/global/one-ones would anyway have to give them explicitly) could be the most clear option.

Even with stream being the default, the one/global cases could still be instantiated from it (the only "penalty" wrt. current situation would be the inclusion of ".../stream/EDProducer.h" header where it is not strictly needed).

Anyway, I'll do an update with default values for template parameters of ObjectSelectorProducer removed.